Networking/Networking/Model/Refund/Refund.swift

```diff
@@ -0,0 +1,167 @@
+    /// When false, manual refund process is used.
+    ///
+    public let isAutomated: Bool?
+
+    /// If true, the automated refund for a payment gateway is used.
+    /// When false, the manual refund process is used.
+    ///
+    public let createAutomated: Bool?
+
+    public let items: [OrderItemRefund]
+
+    /// Refund struct initializer
+    ///
+    public init(refundID: Int,
+                orderID: Int,
+                siteID: Int,
+                dateCreated: Date,
+                amount: String,
+                reason: String,
+                refundedByUserID: Int,
+                isAutomated: Bool?,
+                createAutomated: Bool?,
+                items: [OrderItemRefund]) {
+        self.refundID = refundID
+        self.orderID = orderID
+        self.siteID = siteID
+        self.dateCreated = dateCreated
+        self.amount = amount
+        self.reason = reason
+        self.refundedByUserID = refundedByUserID
+        self.isAutomated = isAutomated
+        self.createAutomated = createAutomated
+        self.items = items
+    }
+
+    // The public initializer for a Refund
+    ///
+    public init(from decoder: Decoder) throws {
+        guard let orderID = decoder.userInfo[.orderID] as? Int else {
+            throw RefundDecodingError.missingOrderID
+        }
+
+        guard let siteID = decoder.userInfo[.siteID] as? Int else {
+            throw RefundDecodingError.missingSiteID
+        }
+
+        let container = try decoder.container(keyedBy: DecodingKeys.self)
+
+        let refundID = try container.decode(Int.self, forKey: .refundID)
+        let dateCreated = try container.decodeIfPresent(Date.self, forKey: .dateCreated) ?? Date()
+        let amount = try container.decode(String.self, forKey: .amount)
+        let reason = try container.decode(String.self, forKey: .reason)
+        let refundedByUserID = try container.decode(Int.self, forKey: .refundedByUserID)
+        let isAutomated = try container.decode(Bool.self, forKey: .automatedRefund)
+        let items = try container.decode([OrderItemRefund].self, forKey: .items)
+
+        self.init(refundID: refundID,
+                  orderID: orderID,
+                  siteID: siteID,
+                  dateCreated: dateCreated,
+                  amount: amount,
+                  reason: reason,
+                  refundedByUserID: refundedByUserID,
+                  isAutomated: isAutomated,
+                  createAutomated: nil,
+                  items: items)
+    }
+
+    // The public initializer for an encodable Refund
+    ///
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: EncodingKeys.self)
+
+        try container.encode(amount, forKey: .amount)
+        try container.encode(reason, forKey: .reason)
+
+        // FIXME: What happens when you encode a struct that has a nested struct?
+        // `items` contains `taxes: [OrderItemTaxRefund]`.
```

Comment:
I think the nested struct with array fields would work? maybe this is something we could verify by unit tests?